driver/src/main/scala/za/co/absa/hyperdrive/driver/IngestionDriver.scala

```diff
@@ -29,11 +30,16 @@ private[driver] abstract class IngestionDriver {
   private val logger = LoggerFactory.getLogger(this.getClass)
   val ListDelimiter = ','
 
+
   def main(args: Array[String]): Unit = {
     logger.info(s"Starting Hyperdrive ${DriverUtil.getVersionString}")
     val configuration = loadConfiguration(args)
+    SecretsConfigUtils.resolveSecrets(configuration)
```

Comment:
Oh, this is terrible, I would prefer to have it as a function but I think you can't do much about it

Reply:
I could `clone` the original config and then call the modifying methods on the clone and return the clone. Not sure if it's worth the effort. I could also just return a reference to the config, but then I pretend it's immutable when it's not.

Reply:
I would leave it now it is more visible that it is mutable